### PR TITLE
Delete Node without Head Pointer in Linked List

### PR DESCRIPTION
**Pull request by anshu189**
Delete Node without head pointer.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: data_structures/delete_node_without_head.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#include <bits/stdc++.h>
```
Comment (mishraabhinn) [inline data_structures/delete_node_without_head.cpp:1]: https://www.geeksforgeeks.org/delete-a-node-from-linked-list-without-head-pointer/
I really feel it copied from here. 
And such a type of code is not acceptable. Please read contributing guidelines.

**Review by mishraabhinn — CHANGES_REQUESTED** [review]

File: data_structures/delete_node_without_head.cpp (hunk 1 above)
Comment (anshu189) [reply data_structures/delete_node_without_head.cpp:1]: Can you please give an overview of what type of changes you want me to be made in this code?

Comment (mishraabhinn) [reply data_structures/delete_node_without_head.cpp:1]: https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md

Comment (anshu189) [reply data_structures/delete_node_without_head.cpp]: Kindly review again, I comitted some necessary changes.

File: data_structures/delete_node_without_head.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
+/**
+ * \file
+ * \brief [Delete a Node Without Head Pointer
+ * in Linked List](https://en.wikipedia.org/wiki/Linked_list#Singly_linked_list)
+ * @author [Anshu Saini](https://github.com/anshu189)
+ */
+
+#include <iostream>
+using namespace std;
```
Comment (Panquesito7) [inline data_structures/delete_node_without_head.cpp:9]: This is discouraged. Please strictly use `std::` for each member.

File: data_structures/delete_node_without_head.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    Node *next;
+    // Constructor for Linked List for a deafult value
+    Node(int data){
+        this -> data=data;
+        this -> next=NULL;
+    }
+};
+
+// Print Linked List Function
+void printList(Node* &head){
```
Comment (Panquesito7) [inline data_structures/delete_node_without_head.cpp:24]: Functions should be documented explaining the use of its parameters, a brief description, and what it returns.

File: data_structures/delete_node_without_head.cpp [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+// Deleting a single Node without head Pointer in Linked List
+// Time Complexity: O(1)
+// Auxilliary Space: O(1)
+void delwohead(Node* del){
+    del->data=del->next->data;
+    del->next=del->next->next;
+}
+
+static void test( Node* head,Node* tail) {
```
Comment (Panquesito7) [inline data_structures/delete_node_without_head.cpp:48]: Preferably, do not use parameters in test functions. Instead, declare them inside the function.
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
static void test( Node* head,Node* tail) {
```

File: data_structures/delete_node_without_head.cpp [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        tempnode=tempnode->next;
+    }
+    delwohead(nodetobedeleted); // Funtion to delete a particular Node
+    cout<<endl<<"<= New Linked List after deleting second element =>"<<endl;
+    
+    printList(head); // Prints out the new Linke List after deletion
+    cout<<"Test case have been successfully passed!"<<endl<<endl;
+}
+
+// Driver Function
```
Comment (Panquesito7) [inline data_structures/delete_node_without_head.cpp:72]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
```

File: data_structures/delete_node_without_head.cpp [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+// Initializing a Linked List using Class Node
+class Node{
+public:
+    int data;
+    Node *next;
+    // Constructor for Linked List for a deafult value
+    Node(int data){
+        this -> data=data;
+        this -> next=NULL;
```
Comment (Panquesito7) [inline data_structures/delete_node_without_head.cpp:19]: `nullptr` should be used instead (same for other instances of the code).

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**